Riot/Modules/Application/LegacyAppDelegate.m

```diff
@@ -1481,42 +1481,28 @@ - (BOOL)handleUniversalLinkWithParameters:(UniversalLinkParameters*)universalLin
                             // Ask the HS to resolve the room alias into a room id and then retry
                             self->universalLinkFragmentPending = fragment;
                             MXKAccount* account = accountManager.activeAccounts.firstObject;
-                            [account.mxSession.matrixRestClient roomIDForRoomAlias:roomIdOrAlias success:^(NSString *roomId) {
+                            [account.mxSession.matrixRestClient resolveRoomAlias:roomIdOrAlias success:^(MXRoomAliasResolution *resolution) {
                                 
                                 // Note: the activity indicator will not disappear if the session is not ready
                                 [homeViewController stopActivityIndicator];
                                 
                                 // Check that 'fragment' has not been cancelled
                                 if ([self->universalLinkFragmentPending isEqualToString:fragment])
                                 {
-                                    // Retry opening the link but with the returned room id
-                                    NSString *newUniversalLinkFragment =
```

Comment:
This is a very messy code, and there is some danger that I am breaking some existing functionality by removing it. But as far as I can see:

1) we are basically just replacing the occurence of an alias by an id in a fragment that pretty much equals the alias to begin with
2) if we get a valid room ID deeplink, this should suffice to join a room no matter what the original fragment contained.

This change could only pose a problem if there are some other non-obvious parts of the room fragment that would be discarded by this.